core/src/main/java/org/elasticsearch/discovery/zen/UnicastZenPing.java

```diff
@@ -377,154 +372,173 @@ public boolean isClosed() {
             return this.closed.get();
         }
 
-        public PingCollection pingCollection() {
-            return pingCollection;
+        public List<DiscoveryNode> getSeedNodes() {
+            checkIfClosed();
+            return seedNodes;
+        }
+
+        public synchronized Connection getConnection(TransportAddress address) {
+            checkIfClosed();
+            return tempConnections.get(address);
+        }
+
+        private void checkIfClosed() {
+            if (isClosed()) {
+                throw new AlreadyClosedException("pinging round [" + id + "] is finished");
+            }
+        }
+
+        public synchronized Connection addConnectionIfNeeded(TransportAddress address, Connection newConnection) {
```

Comment:
turns out we do need this protection or something similar. I took another approach, which I think you'd like better.